tfhe/src/shortint/server_key/shift.rs

```diff
@@ -762,15 +759,21 @@ impl ServerKey {
     /// assert_eq!(msg << shift, msg_and_carry);
     /// assert_eq!((msg << shift) % modulus, msg_only);
     /// ```
+    #[allow(clippy::needless_pass_by_ref_mut)]
     pub fn smart_scalar_left_shift(&self, ct: &mut Ciphertext, shift: u8) -> Ciphertext {
-        ShortintEngine::with_thread_local_mut(|engine| {
-            engine.smart_scalar_left_shift(self, ct, shift)
-        })
+        let mut result = ct.clone();
+        self.smart_scalar_left_shift_assign(&mut result, shift);
+        result
```

Comment:
do we want to update this to clean the input ciphertext ? can be done a follow up PR/issue

Reply:
https://github.com/zama-ai/tfhe-rs-internal/issues/353

Reply:
Yes, I'd prefer in a follow up PR
Thanks for this issue!

Reply:
of course let's not load this PR more !